src/core/DataFile.cpp

```diff
@@ -163,6 +163,53 @@ DataFile::~DataFile()
 
 
 
+bool DataFile::validate( QString extension )
+{
+	switch( m_type )
+	{
+	case Type::SongProject:
+		if( extension == "mmp" || extension == "mmpz" )
+		{
+			return true;
+		}
+		break;
+	case Type::SongProjectTemplate:
+		if(  extension == "mpt" )
+		{
+			return true;
+		}
+		break;
+	case Type::InstrumentTrackSettings:
```

Comment:
@softrabbit @Spekular @curlymorphic I think we need an `XIZ` check in here somewhere...  :)